python/lsst/pipe/tasks/multiBand.py

```diff
@@ -818,14 +818,271 @@ def getSkySourceFootprints(self, mergedList, skyInfo, seed):
         return converted
 
 
+class DeblendCoaddSourcesConfig(Config):
+    """DeblendCoaddSourcesConfig
+
+    Configuration parameters for the `DeblendCoaddSourcesTask`.
+    """
+    singleBandDeblend = ConfigurableField(target=SourceDeblendTask,
+                                          doc="Deblend sources separately in each band")
+    multiBandDeblend = ConfigurableField(target=MultibandDeblendTask,
+                                         doc="Deblend sources simultaneously across bands")
+    simultaneous = Field(dtype=bool, default=True, doc="Simultaneously deblend all bands?")
+    coaddName = Field(dtype=str, default="deep", doc="Name of coadd")
+
+
+class DeblendCoaddSourcesRunner(TaskRunner):
+    """Get the psfCache setting into DeblendCoaddSourcesTask"""
+    @staticmethod
+    def getTargetList(parsedCmd, **kwargs):
+        """!
+        @brief Provide a list of patch references for each patch.
+
+        The patch references within the list will have different filters.
+
+        @param[in]  parsedCmd  the parsed command
+        @param      **kwargs   key word arguments (unused)
+        @throws RuntimeError if multiple references are provided for the same combination of tract, patch and
+        filter
+        """
+        refList = {}  # Will index this as refList[tract][patch][filter] = ref
+        for ref in parsedCmd.id.refList:
+            tract = ref.dataId["tract"]
+            patch = ref.dataId["patch"]
+            filter = ref.dataId["filter"]
+            if tract not in refList:
+                refList[tract] = {}
+            if patch not in refList[tract]:
+                refList[tract][patch] = {}
+            if filter in refList[tract][patch]:
+                raise RuntimeError("Multiple versions of %s" % (ref.dataId,))
+            refList[tract][patch][filter] = ref
+        kwargs["psfCache"] = parsedCmd.psfCache
+        return [(list(p.values()), kwargs) for t in refList.values() for p in t.values()]
+
+    def makeTask(self, parsedCmd=None, args=None):
+        """A variant of the base version that passes a butler argument to the task's constructor.
+
+        Parameters
+        ----------
+        parsedCmd : `argparse.Namespace`
+            Parsed command-line options,
+            as returned by the `~lsst.pipe.base.ArgumentParser`;
+            if specified then args is ignored.
+        args
+            Other arguments; if ``parsedCmd`` is `None` then this must be specified.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if ``parsedCmd`` and ``args`` are both `None`.
+        """
+        if parsedCmd is not None:
+            butler = parsedCmd.butler
+        elif args is not None:
+            dataRefList, kwargs = args
+            butler = dataRefList[0].getButler()
+        else:
+            raise RuntimeError("Neither parsedCmd or args specified")
+        return self.TaskClass(config=self.config, log=self.log, butler=butler)
```

Comment:
Seems like you could just use `MergeSourcesRunner` instead of writing your own, *if* you made `psfCache` a config parameter (which might be a good idea anyway, rather than making it special - though I can see how you might want that, given that it's not an option that should change the outputs).

Even if you do want to keep `psfCache` as a special command-line argument, I'm hoping you could inherit from `MergeSourcesRunner` and eliminate most of this code.

Reply:
What is `psfCache`? I just added it because it's something that `MeasureMergedCoaddSources` had, but I don't necessarily know what it does, and if it's worth keeping.

Reply:
Ah, I bet I know, but my knowledge of the details is sufficiently limited that we should ask @PaulPrice.

Reply:
It's the `lsst::utils::Cache` embedded in `lsst::afw::detection::Psf` that caches the most recent PSF image computations. Changing the capacity of the cache influences the job runtime and memory usage.